src/main/java/com/conveyal/r5/transit/TransitLayer.java

```diff
@@ -682,18 +693,23 @@ public static TransitModes getTransitModes(int routeType) {
     /**
      * @return a semi-shallow copy of this transit layer for use when applying scenarios.
      */
-    public TransitLayer scenarioCopy(TransportNetwork newScenarioNetwork) {
+    public TransitLayer scenarioCopy(TransportNetwork newScenarioNetwork, boolean willBeModified) {
```

Comment:
Yes. I added some comments here. It's questionable whether this optimization actually affects routing speed, but in theory it saves a comparison and an extra dereference every time we use the edge/vertex stores.